ktor-server/ktor-server-tests/jvm/test/io/ktor/server/http/spa/SinglePageApplicationTest.kt

```diff
@@ -161,17 +194,23 @@ class SinglePageApplicationTest {
             routing {
                 singlePageApplication {
                     useResources = true
-                    filesPath = "io.ktor.server.http.spa"
-                    defaultPage = "SinglePageApplicationTest.class"
+                    filesPath = "public"
+                    defaultPage = "default.txt"
                     ignoreFiles { true }
                 }
             }
         }
 
-        assertEquals(HttpStatusCode.OK, client.get("/SinglePageApplicationTest.class").status)
-        assertEquals(HttpStatusCode.OK, client.get("/Empty1.class").status)
-        assertEquals(HttpStatusCode.OK, client.get("/a").status)
-        assertEquals(HttpStatusCode.OK, client.get("/").status)
+        assertEquals(HttpStatusCode.Forbidden, client.get("/file.txt").status)
+
+        client.get("/a").let {
+            assertEquals(it.status, HttpStatusCode.OK)
```

Comment:
Shouldn't there be a `HttpStatusCode.Forbidden` because of `ignoreFiles { true }`? In `testIgnoreAllRoutes` on 90[new] line there is something similar and there is `HttpStatusCode.Forbidden`

Reply:
Resources work a bit differently from files here. We can't check for `ignoreFiles` for resources that do not exist since we can't get `URL` for them. But we can reply with default resource.

Reply:
It's inconsistent, but it's a weird case, so I don't think that it will be a problem.